nupic/algorithms/anomaly.py

```diff
@@ -90,26 +91,26 @@ def __init__(self, slidingWindowSize=None, mode=MODE_PURE, binaryAnomalyThreshol
     @param binaryAnomalyThreshold (optional) - if set [0,1] anomaly score
          will be discretized to 1/0 (1 if >= binaryAnomalyThreshold)
          The transformation is applied after moving average is computed and updated.
+         0.0 means disabled (default)
     """
     self._mode = mode
-    if slidingWindowSize is not None:
+    if slidingWindowSize > 0:
       self._movingAverage = MovingAverage(windowSize=slidingWindowSize)
     else:
       self._movingAverage = None
 
-    if self._mode == Anomaly.MODE_LIKELIHOOD or self._mode == Anomaly.MODE_WEIGHTED:
+    if self._mode == Anomaly.MODE_LIKELIHOOD or \
+       self._mode == Anomaly.MODE_WEIGHTED:
       self._likelihood = AnomalyLikelihood() # probabilistic anomaly
-    if not self._mode in Anomaly._supportedModes:
-      raise ValueError("Invalid anomaly mode; only supported modes are: "
-                       "Anomaly.MODE_PURE, Anomaly.MODE_LIKELIHOOD, "
-                       "Anomaly.MODE_WEIGHTED; you used: %r" % self._mode)
+    assert self._mode in Anomaly._supportedModes, "Invalid anomaly mode; \
+                       only supported modes are: \n\
+                       Anomaly.MODE_PURE, Anomaly.MODE_LIKELIHOOD, \n\
+                       Anomaly.MODE_WEIGHTED;\n you used: %r" % self._mode
     self._binaryThreshold = binaryAnomalyThreshold
-    if binaryAnomalyThreshold is not None and ( 
-          not isinstance(binaryAnomalyThreshold, float) or
-          binaryAnomalyThreshold >= 1.0  or 
-          binaryAnomalyThreshold <= 0.0 ):
-      raise ValueError("Anomaly: binaryAnomalyThreshold must be from (0,1) "
-                       "or None if disabled.")
+    assert isinstance(binaryAnomalyThreshold, float) and \
```

Comment:
`isinstance(binaryAnomalyThreshold, float)` will always be true because you changed the default value to `0.0`.